build/async.js

```diff
@@ -0,0 +1,1225 @@
+'use strict';
```

Comment:
This is the UMD build.  I called it `async.js` because it is the most analagous to the `lib/async.js` in v1.x and prior.  The CJS build is called `index.js`.

I'm leaving the `dist/` folder in the root so it can be easily targeted by bower users.  Moving it to `build/dist/` is no problem.

The `es/` folder is an experiment.  I'm copying what lodash 4.x does with `lodash/fp`.  You can `require('async/waterfall')` or `import waterfall from 'async/es/waterfall'` with the same package.